app/parse.py

```diff
@@ -15,9 +46,85 @@ class Product:
     num_of_reviews: int
 
 
+def parse_single_product(product_soup: Tag) -> Product:
+    return Product(
+        title=product_soup.select_one(".title")["title"],
+        description=product_soup.select_one(
+            ".description").text.replace("\xa0", " "),
+        price=float(product_soup.select_one(".price").text.replace("$", "")),
+        rating=len(product_soup.select(".glyphicon-star")),
+        num_of_reviews=int(
+            product_soup.select_one(".ratings > p.pull-right").text.split()[0]
+        ),
+    )
+
+
+def check_more_pages(page_soup: BeautifulSoup) -> str:
+    more_pages = ""
+    try:
+        more_pages = page_soup.select_one(".ecomerce-items-scroll-more").text
+    except AttributeError:
+        pass
+    return more_pages
+
+
+def get_all_page_products(page_soup: BeautifulSoup) -> list:
```

Comment:
add additional typehint for list inner objects like list[YOUR_VALUE_TYPE]
for example: list that contain string elements has typehint list[str]